components/use-crossposter.js

```diff
@@ -20,14 +38,81 @@ async function discussionToEvent (item) {
   }
 }
 
+async function linkToEvent (item) {
+  const createdAt = Math.floor(Date.now() / 1000)
+
+  return {
+    created_at: createdAt,
+    kind: 1,
+    content: `${item.title} \n ${item.url}`,
+    tags: []
+  }
+}
+
+async function pollToEvent (item) {
+  const createdAt = Math.floor(Date.now() / 1000)
+
+  const expiresAt = createdAt + 86400
+
+  return {
+    created_at: createdAt,
+    kind: 1,
+    content: item.text,
+    tags: [
+      ['poll', 'single', expiresAt.toString(), item.title, ...item.poll.options.map(op => op?.option.toString())]
+    ]
+  }
+}
+
+async function bountyToEvent (item) {
+  const createdAt = Math.floor(Date.now() / 1000)
+
+  return {
+    created_at: createdAt,
+    kind: 30402,
+    content: item.text,
+    tags: [
+      ['d', item.id.toString()],
+      ['title', item.title],
+      ['location', `https://stacker.news/items/${item.id}`],
+      ['price', item.bounty.toString(), 'SATS'],
+      ['t', 'bounty'],
+      ['published_at', createdAt.toString()]
+    ]
+  }
+}
+
 export default function useCrossposter () {
-  const toast = useToast()
+  const toaster = useToast()
   const { data } = useQuery(SETTINGS)
-  const relays = [...DEFAULT_CROSSPOSTING_RELAYS, ...(data?.settings?.nostrRelays || [])]
+  const userRelays = data?.settings?.privates?.nostrRelays || []
+  const relays = [...DEFAULT_CROSSPOSTING_RELAYS, ...userRelays]
+
+  const [fetchItem] = useLazyQuery(
+    gql`
+      ${ITEM_FULL_FIELDS}
+      ${POLL_FIELDS}
+      query Item($id: ID!) {
+        item(id: $id) {
+          ...ItemFullFields
+          ...PollFields
+        }
+      }`
+  )
```

Comment:
You need to add `fetchPolicy: 'no-cache'` here.[^1] Else, on edits, this will returned the cached item and thus the same item will be crossposted.

[^1]: https://www.apollographql.com/docs/react/data/queries/#supported-fetch-policies